src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -237,13 +240,16 @@
           [:centralized-metrics/track :metric/swap-proposal-failed {:error (:code error-response)}]]]}))
 
 (rf/reg-event-fx :wallet/stop-get-swap-proposal
- (fn []
-   {:json-rpc/call [{:method   "wallet_stopSuggestedRoutesAsyncCalculation"
-                     :params   []
-                     :on-error (fn [error]
-                                 (log/error "failed to stop fetching swap proposals"
-                                            {:event :wallet/stop-get-swap-proposal
-                                             :error error}))}]}))
+ (fn [{:keys [db]}]
+   (let [last-request-uuid (get-in db [:wallet :ui :swap :last-request-uuid])]
+     (when last-request-uuid
```

Comment:
If last-request-uuid exists, it means there is a swap route request ongoing and we can call `wallet_stopSuggestedRoutesAsyncCalculation` to stop receiving signals. Otherwise, there's no need to call the endpoint.

Reply:
The comment you provided can be also reflected in code by renaming variable:
```clojure
(let [route-request-ongoing? (some? (get-in db [:wallet :ui :swap :last-request-uuid]))]
     (route-request-ongoing?
     ...
```